src/MorphoAaveV3Adapter.ts

```diff
@@ -713,43 +704,43 @@ export class MorphoAaveV3Adapter extends MorphoAaveV3DataEmitter {
 
         if (fetch) {
           await Promise.all([
-            this._marketFetcher.fetchMarketData(
-              underlyingAddress,
-              {
-                priceSource:
-                  isEmode && priceSource !== constants.AddressZero
-                    ? priceSource
-                    : underlyingAddress,
-              },
-              blockTag
-            ),
-            this._marketSupplyFetcher.fetchMarketSupply(
-              underlyingAddress,
-              blockTag
-            ),
-          ]).then(([marketData, marketSupplyData]) => {
-            if (
-              !validateMarketSupplyData(
-                marketSupplyData,
-                marketData,
-                marketConfig
+            this._marketFetcher
+              .fetchMarketData(
+                underlyingAddress,
+                {
+                  priceSource:
+                    isEmode && priceSource !== constants.AddressZero
+                      ? priceSource
+                      : underlyingAddress,
+                },
+                blockTag
               )
-            ) {
-              marketSupplyData = {
-                scaledMorphoCollateral: constants.Zero,
-                scaledMorphoSupplyOnPool: constants.Zero,
-              };
-            }
-
-            // We first add the scaled balances
-            this._scaledMarketsData[underlyingAddress] = {
-              ...marketData,
-              ...marketSupplyData,
-            };
-          });
+              .then((marketData) => {
+                const isCollateralOnPool =
+                  getAddress(underlyingAddress) !== getAddress(addresses.weth);
```

Comment:
Would be better to just check if the market is an eMode market imo (no need to change again if we list a new asset in eMode

Reply:
mmh not really, stEth and rEth are in emode, this is only eth that has supply enabled